waku/v2/node/rest/store/handlers.nim

```diff
@@ -0,0 +1,251 @@
+when (NimMajor, NimMinor) < (1, 4):
+  {.push raises: [Defect].}
+else:
+  {.push raises: [].}
+
+import
+  stew/results,
+  chronicles,
+  json_serialization,
+  presto/route
+import
+  ../../../../common/base64,
+  ../../../protocol/waku_message/topics/content_topic,
+  ../../../protocol/waku_store/common,
+  ../../../utils/time,
+  ../../waku_node,
+  ../../peer_manager,
+  ../responses,
+  ../serdes,
+  ./types
+
+export types
+
+logScope:
+  topics = "waku node rest store_api"
+
+const futTimeout* = 5.seconds # Max time to wait for futures
+
+# Queries the store-node with the query parameters and
+# returns a RestApiResponse that is sent back to the api client.
+proc makeHistoryQuery(selfNode: WakuNode,
+                      histQuery: HistoryQuery,
+                      storePeer: RemotePeerInfo):
+
+                      Future[RestApiResponse] {.async.} =
+
+  let queryFut = selfNode.query(histQuery, storePeer)
+  if not await queryFut.withTimeout(futTimeout):
+    const msg = "No history response received (timeout)"
+    error msg
+    return RestApiResponse.internalServerError(msg)
+
+  let res = queryFut.read()
+  if res.isErr():
+    const msg = "Error occurred in queryFut.read()"
+    error msg, error=res.error
+    return RestApiResponse.internalServerError(msg)
+
+  let storeResp = res.value.toStoreResponseRest()
+  let resp = RestApiResponse.jsonResponse(storeResp, status=Http200)
+  if resp.isErr():
+    const msg = "Error building the json respose"
+    error msg, error=resp.error
+    return RestApiResponse.internalServerError(msg)
+
+  return resp.get()
+
+# Converts timestamp in string format to Timestamp object
+proc toTimestamp(input: string): Timestamp =
+  try:
+    return Timestamp(parseInt(input))
+  except CatchableError as e:
+    error "Error converting to Timestamp ", error=e.msg
```

Comment:
Yes I agree, I will pick `debug` level in this case.
If this exception occurs, the problem comes from how the user sent the parameter, rather than an app error, so it is important to give good feedback to the user and just log in `debug`.